examples/cd_player.py

```diff
@@ -0,0 +1,174 @@
+        """ Respond to 'tick' event while we are in each state """
+        logger.info(f'Received tick in state {state.name}')
+
+    def register_handlers(self):
+        self.handlers = {'enter': self._on_enter,
+                         'exit': self._on_exit,
+                         'tick': self._on_tick}
+
+
+class HybridSuperState(pysm.StateMachine):
+    """ Super-states (composite states) that know how to interact with the controller """
+
+    def __init__(self, controller):
+        super().__init__(self.__class__.__name__)
+        # keep an instance of the controller so that we can actually do things
+        self._controller = controller
+
+    def _on_enter(self, state, event):
+        """ Default entry action is just to log the entry """
+        logger.info(f'Entered state {state.name}')
+
+    def _on_exit(self, state, event):
+        """ Default exit action is just to log the exit"""
+        logger.info(f'Exited state {state.name}')
+
+    def _on_tick(self, state, event):
+        """ Respond to 'tick' event while we are in each state """
+        logger.info(f'Received tick in state {state.name}')
+
+    def register_handlers(self):
+        self.handlers = {'enter': self._on_enter,
+                         'exit': self._on_exit,
+                         'tick': self._on_tick}
+
+
+class CdPlayer(HybridSuperState):
+    """ The one-and-only top-level state (machine) """
+
+    def __init__(self, controller):
+        super().__init__(controller)
+
+        self.add_state(NotBroken(controller), initial=True)
+        self.add_state(Broken(controller))
+
+        # I believe that this is interpreted with local (as opposed to external) transition syntax
+        # https://en.wikipedia.org/wiki/UML_state_machine#Local_versus_external_transitions
+        # TODO: Consider modifying psym library to support this
+        self.add_transition(from_state=self, to_state=self.substate('Broken'), events=[CdEvent.HAMMER])
+
+        # Final step
+        self.initialize()
+
+    def dispatch(self, event):
+        logger.info(f'Dipatching {event}...')
+        super().dispatch(pysm.Event(event))
+
+    def tick(self):
+        logger.info('Injecting tick...')
+        super().dispatch(pysm.Event('tick'))
+
+
+class Broken(HybridSuperState):
+    pass
+
+
+class NotBroken(HybridSuperState):
+    pass
+
+    def __init__(self, controller):
+        super().__init__(controller)
+        self.add_state(Stopped(controller), initial=True)
+        self.add_state(Open(controller))
+        self.add_state(Empty(controller))
+        self.add_state(Playing(controller))
+        self.add_state(Paused(controller))
+
+        # I would MUCH rather define these transitions as part of the origin state's class!
+        # This feels like I have to define everything one level higher than I would like.
+        self.add_transition(from_state=self.substate('Stopped'), to_state=self.substate('Playing'), events=[CdEvent.PLAY])
```

Comment:
There are a couple of things going on.  
* I can get behind defining transitions along with the state that they originate with (i.e. what events does this particular state know how to handle?) or globally in one big table for the whole state machine.  This is currently the worst of both worlds where transitions are defined at the level of the state containing the source state.  
* I need to confirm whether or not pysm will let you define a transition to a state that is outside of the current state.
* Presently, the transition behavior depends (incorrectly) on what composite state you define the transition in (not just the states that it is from and to) because the entry/exit sequence crawls up the tree to the **definition** of the transition, not to the least-common-ancesstor. (#3)

I'm going to work on fixing #3 and then we will be able to choose where we want the definitions to live on a case-by-case basis.

Reply:
Regarding your 2nd bullet, in the diagram below from the pysm documentation it looks like they have a transition from s11 to s211. Or is that not what you mean, since they're both contained by s0?
![image](https://github.com/nealtanner/pysm/assets/78115203/209ea274-3b7b-4e87-aaea-0b0b6a8d5e21)